### PR TITLE
PUBDEV-6948: Disable HTTP TRACE Method

### PR DESCRIPTION
**Pull request by michalkurka**
https://0xdata.atlassian.net/browse/PUBDEV-6948

### CODE REVIEW

**Comment by michalkurka** [comment]
cc: @mn-mikke - we will need to make a nightly build of SW that bundles this fix to test with an user

File: h2o-core/src/test/java/water/webserver/H2OHttpViewImplTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    };
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+    
+    when(request.getMethod()).thenReturn("TRACE");
+    
+    httpView.gateHandler(request, response);
+
+    verify(response).setStatus(405);
```
Comment (Pscheidl) [inline h2o-core/src/test/java/water/webserver/H2OHttpViewImplTest.java]: Great. 405 is the correct answer indeed.

File: h2o-core/src/main/java/water/server/ServletUtils.java [hunk 2]
```diff
@@ -172,6 +174,10 @@ public static boolean isXhrRequest(final HttpServletRequest request) {
     return "XMLHttpRequest".equals(requestedWithHeader);
   }
 
+  public static boolean isTraceRequest(final HttpServletRequest request) {
+    return TRACE_METHOD.equalsIgnoreCase(request.getMethod());
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/server/ServletUtils.java]: Ignoring case is necessary. Some implementations actually might screw up the case. I've seen this many times.

**Comment by Pscheidl** [comment]
Re-started the tests manually. No PR-related failures observed, one failure due to infrastructure.

**Comment by mn-mikke** [comment]
> we will need to make a nightly build of SW that bundles this fix to test with an user
@michalkurka No problem, just let me know once you have the nightly H2O-3 build.

**Comment by Pscheidl** [comment]
Went through the test failures - unrelated.

**Pull request merged**